### PR TITLE
docs(functions): Added How Location/Region Cen be Specified

### PR DESCRIPTION
**Pull request by Bilal-Abdeen**
### Description

Cloud functions can be deployed in various locations/regions. Specifying the location was not documented. So, I added it to the documentation. 

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [X ] Yes
- My change supports the following platforms;
  - [ X] `Android`
  - [ X] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ X] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Review by mikehardy — CHANGES_REQUESTED** [review]
This adds a lot of "conditional thinking" to the documentation ("do this, or if you want this do this other thing")

I believe it would be better to have a separate "region-specific" section where it was separately discussed, something like (after a minor header) "If you deploy your functions to a specific region or want to call your functions in a specific region, you need to get the functions instance in a different way. The way to get a functions instance specific to a region is .... `code here`" 

What do you think?

**Comment by Bilal-Abdeen** [comment]
@mikehardy 
I am sorry for not being familiar with the procedure to get PRs across the line. I thought this was approved for merging!? Is there anything I need to do?

**Comment by mikehardy** [comment]
Looks like it just needs my attention sorry for the delay. I've switched development machine environment completely and have had a real struggle getting productive again, doesn't excuse the delay but hopefully explains that it was nothing to do with your contribution! Queued for review and I've got a batch of stuff it'll go with in the next few days

File: docs/functions/usage/index.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Whilst developing your application with Cloud Functions, it is possible to run t
 To call the emulated functions, call the `useEmulator` method exposed by the library:
 
 ```js
-import functions from '@react-native-firebase/functions';
+import { firebase } from '@react-native-firebase/functions';
```
Comment (mikehardy) [inline docs/functions/usage/index.md:57]: This is not correct - it does not match the code below where `functions()` is usesd

File: docs/functions/usage/index.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ Assuming we have a deployed a callable endpoint named `listProducts`, to call th
 
 ```js
 // Deployed HTTPS callable
+
```
Comment (mikehardy) [inline docs/functions/usage/index.md:73]:
```suggestion
```

Please do not change unrelated spacing in PRs

Comment (mikehardy) [inline docs/functions/usage/index.md:73]:
```suggestion
```

Please do not change unrelated spacing in PRs

File: docs/functions/usage/index.md [hunk 3]
```diff
@@ -89,7 +90,7 @@ function App() {
 
   useEffect(() => {
     functions()
-      .httpsCallable('listProducts')()
+      .httpsCallable('listProducts')({ abc: 123 })
```
Comment (mikehardy) [inline docs/functions/usage/index.md:93]: This is changed with `{ abc: 123}` added but no explanation for the change. Could use a comment in the code perhaps?

File: docs/functions/usage/index.md [hunk 4]
```diff
@@ -103,3 +104,22 @@ function App() {
...
+If you need to deploy Functions in a region other than the default one, modified statements need to be used. 
+
+```javascript
+import { firebase } from '@react-native-firebase/functions';
+  
+  // Emulator statement 
+  firebase.app().functions("region_name").useEmulator('localhost', 5001);
+
+  // definition statement 
+  functions.region("region_name").https.onCall(() => {});
```
Comment (mikehardy) [inline docs/functions/usage/index.md:119]: Definition where? Where is this documented upstream? Could use a pointer. I'm not familiar with a need to define a function in a react-native app, I'm only familiar with defining functions in separate modules that are built + installed by the firebase tools.

**Review by mikehardy — CHANGES_REQUESTED** [review]
I have some questions / thoughts on this one

File: docs/functions/usage/index.md (hunk 1 above)
Comment (Bilal-Abdeen) [reply docs/functions/usage/index.md]: You are right. I reversed the change.

File: docs/functions/usage/index.md (hunk 3 above)
Comment (Bilal-Abdeen) [reply docs/functions/usage/index.md:92]: You are right. I removed the addition.

File: docs/functions/usage/index.md (hunk 4 above)
Comment (Bilal-Abdeen) [reply docs/functions/usage/index.md]: You are right. I removed the addition.

File: docs/functions/usage/index.md [hunk 5]
```diff
@@ -89,7 +89,7 @@ function App() {
 
   useEffect(() => {
     functions()
-      .httpsCallable('listProducts')()
+      .httpsCallable('listProducts')({ abc: 123 })
```
Comment (mikehardy) [inline docs/functions/usage/index.md:92]: This change is still present, I'm proposing a suggestion and will commit it - the rest looks great now

```suggestion
      .httpsCallable('listProducts')()
```

**Review by mikehardy — APPROVED** [review]
This looks like a nice addition, thank you!

**Review by mikehardy — APPROVED** [review]
I fixed the lint error via `yarn lint:markdown --write` and repushed

**Pull request merged**